lib/compat/wordpress-6.7/post.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Temporary compatibility shims for block APIs present in Gutenberg.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Add the default_rendering_mode property to the WP_Post_Type object.
+ * This property can be overwritten by using the post_type_default_rendering_mode filter.
+ *
+ * @param array  $args      Array of post type arguments.
+ * @return array Updated array of post type arguments.
+ */
+function gutenberg_post_type_default_rendering_mode( $args ) {
+	if (
+		( isset( $args['show_in_rest'] ) && $args['show_in_rest'] ) &&
+		( isset( $args['supports'] ) && in_array( 'editor', $args['supports'], true ) ) &&
+		( ! isset( $args['default_rendering_mode'] ) )
+	) {
+		$args['default_rendering_mode'] = 'post-only';
```

Comment:
I think that makes sense to do, @ramonjd -- What are all the value values here? Is it just `post-only`, `template-only`, `template-lock`?

Reply:
It sounds like there's an [agreement](https://github.com/WordPress/gutenberg/pull/62304#pullrequestreview-2102389815) that `template-lock` should be the default, so looks like things work out of the box, even if unintentionally 😄 

However I'd say it'd be beneficial to be intentional about which values are acceptable to avoid state pollution or any side-effects on the frontend.

> Is it just post-only, template-only, template-lock?

If these docs are accurate, then the valid values are `post-only` and `template-locked`

https://github.com/WordPress/gutenberg/blob/3bd67998aec448cfb3e663895def681d21648181/docs/reference-guides/data/data-core-editor.md#L1451-L1451

Reply:
Updated! It seems `template-lock` and `template-locked` are both actually supported, so I've added support for both in the check.